site/utils/index.ts

```diff
@@ -55,3 +59,13 @@ export const zodDate = (defaultError?: string) =>
 
 export const zodTime = (defaultError?: string) =>
     z.string(defaultError ? setZodDefaultError(defaultError) : {}).regex(timeRegex);
+
+export const flattenZodErrors = (errors: ZodError) =>
+    Object.values(
+        errors.flatten<ErrorInfo>((val) => ({
+            errorMessage: val.message,
+            id: val.path[0].toString(),
```

Comment:
There's a bug with the "At least one validity period must be provided" showing since zod is prepending validity into the path array, we should be able to get around this by changing this line to `id: val.path.at(-1)?.toString() ?? "",` so that it grabs the last item in the path array rather than the first